### PR TITLE
Feat/baggage upload

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Add new column luggage_description to entity and update data for seeds.
- Create new module for uploading, deletion and fetching images.

photo_2024-12-23_20-49-10

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-181)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

- [x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [x] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

### CODE REVIEW

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/common/database/entities/luggage.entity.ts [hunk 1]
```diff
@@ -15,6 +15,9 @@ export class Luggage {
   @Column({ nullable: false })
   luggage_weight: number;
 
+  @Column({ nullable: false })
```
Comment (Parle648) [inline src/common/database/entities/luggage.entity.ts:18]: updated entity should be in the PR description

File: src/modules/luggage-images/luggage-images.controller.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import { UploadImageDto } from './dto/upload-image.dto';
+import { LuggageImagesService } from './luggage-images.service';
+
+@ApiTags('Luggage Images')
+@Controller('luggage-images')
+export class LuggageImagesController {
+  constructor(private readonly luggageImagesService: LuggageImagesService) {}
+
+  @Get('luggage/:orderId')
```
Comment (Parle648) [inline src/modules/luggage-images/luggage-images.controller.ts]: idk for me get request to luggage/1 should return luggage which id is equal 1. Maybe luggage/get-by-order or smth like this

File: src/modules/luggage-images/luggage-images.controller.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          cb(null, uniqueName);
+        },
+      }),
+    }),
+  )
+  async uploadImage(@UploadedFile() file: Express.Multer.File, @Body() uploadImageDto: UploadImageDto): Promise<string> {
+    return this.luggageImagesService.uploadImage(file, uploadImageDto.luggageId);
+  }
+
+  @Delete('luggage/:luggageId/images')
```
Comment (Parle648) [inline src/modules/luggage-images/luggage-images.controller.ts]: luggage-images/:luggageId for me looks better

File: src/common/database/entities/luggage.entity.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/common/database/entities/luggage.entity.ts:18]: screen added

File: src/modules/luggage-images/luggage-images.controller.ts (hunk 2 above)
Comment (oyakovytskyi) [reply src/modules/luggage-images/luggage-images.controller.ts]: changed to luggage/get-by-order/:orderId

File: src/modules/luggage-images/luggage-images.controller.ts (hunk 3 above)
Comment (oyakovytskyi) [reply src/modules/luggage-images/luggage-images.controller.ts]: changed to :luggageId

**Review by Parle648 — APPROVED** [review]

**Review by AlexeyBezrebriy — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**